src/nplinker/scoring/link_graph.py

```diff
@@ -267,6 +258,101 @@ def filter(self, u_nodes: Sequence[Entity], v_nodes: Sequence[Entity] = [], /) -
 
         return lg
 
+    @validate_uv
+    def get_link_data(
+        self,
+        u: Entity,
+        v: Entity,
+    ) -> LINK_DATA | None:
+        """Get the data for a link between two objects.
+
+        Args:
+            u: the first object, either a GCF, Spectrum, or MolecularFamily
+            v: the second object, either a GCF, Spectrum, or MolecularFamily
+
+        Returns:
+            A dictionary of scoring methods and their data for the link between the two objects, or
+            None if there is no link between the two objects.
+
+        Examples:
+            >>> lg.get_link_data(gcf, spectrum)
+            {"metcalf": Score("metcalf", 1.0, {"cutoff": 0.5})}
+        """
+        return self._g.get_edge_data(u, v)  # type: ignore
+
+    def get_table_data(self, display_limit: int | None = None) -> list[dict[str, Any]]:
+        """Generate the table data for the LinkGraph.
+
+        This method iterates over the links in the LinkGraph and constructs a table
+        containing information about genomic and metabolomic objects, as well as their
+        associated scores. Each row in the table represents a link between a genomic
+        object and a metabolomic object.
+
+        Args:
+            display_limit (int | None): The maximum number of rows to include in the
+                table. If None, all rows are included.
+
+        Returns:
+            A list of dictionaries containing the table data.
+        """
+        table_data = []
+        for index, link in enumerate(self.links, start=1):
+            table_data.append(self.link_to_dict(link, index))
+            if display_limit is not None and index == display_limit:
+                break
+        return table_data
+
+    @validate_uv
+    def has_link(self, u: Entity, v: Entity) -> bool:
+        """Check if there is a link between two objects.
+
+        Args:
+            u: the first object, either a GCF, Spectrum, or MolecularFamily
+            v: the second object, either a GCF, Spectrum, or MolecularFamily
+
+        Returns:
+            True if there is a link between the two objects, False otherwise
+
+        Examples:
+            >>> lg.has_link(gcf, spectrum)
+            True
+        """
+        return self._g.has_edge(u, v)
+
+    @staticmethod
+    def link_to_dict(link: LINK, index: int) -> dict[str, Any]:
```

Comment:
`index` is not necessary for this method, and it can be added to the function which really requires it.